kotlinpoet/src/jvmMain/kotlin/com/squareup/kotlinpoet/Util.jvm.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.kotlinpoet
+
+import com.squareup.kotlinpoet.CodeBlock.Companion.isPlaceholder
+import java.util.Collections
+import java.util.EnumSet
+
+internal actual fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V> =
+  Collections.unmodifiableMap(LinkedHashMap(this))
+
+internal actual fun <T> Collection<T>.toImmutableList(): List<T> =
+  Collections.unmodifiableList(ArrayList(this))
+
+internal actual fun <T> Collection<T>.toImmutableSet(): Set<T> =
+  Collections.unmodifiableSet(LinkedHashSet(this))
+
+internal actual fun formatIsoControlCode(code: Int): String =
+  String.format("\\u%04x", code)
+
+internal actual fun Int.toHexStr(): String =
+  Integer.toHexString(this)
+
+// TODO Waiting for `CodeBlock` migration.
+internal fun CodeBlock.ensureEndsWithNewLine() = trimTrailingNewLine('\n')
+
+// TODO Waiting for `CodeBlock` migration.
+internal fun CodeBlock.trimTrailingNewLine(replaceWith: Char? = null) = if (isEmpty()) {
+  this
+} else {
+  with(toBuilder()) {
+    val lastFormatPart = trim().formatParts.last()
+    if (lastFormatPart.isPlaceholder && args.isNotEmpty()) {
+      val lastArg = args.last()
+      if (lastArg is String) {
+        val trimmedArg = lastArg.trimEnd('\n')
+        args[args.size - 1] = if (replaceWith != null) {
+          trimmedArg + replaceWith
+        } else {
+          trimmedArg
+        }
+      }
+    } else {
+      formatParts[formatParts.lastIndexOf(lastFormatPart)] = lastFormatPart.trimEnd('\n')
+      if (replaceWith != null) {
+        formatParts += "$replaceWith"
+      }
+    }
+    return@with build()
+  }
+}
+
+private val IDENTIFIER_REGEX = IDENTIFIER_REGEX_VALUE.toRegex()
+
+internal actual val String.isIdentifier: Boolean
+  get() = IDENTIFIER_REGEX.matches(this)
+
+internal actual inline fun <reified E : Enum<E>> enumSetOf(vararg values: E): MutableSet<E> {
+  return when (values.size) {
+    0 -> EnumSet.noneOf(E::class.java)
+    1 -> EnumSet.of(values[0])
+    2 -> EnumSet.of(values[0], values[1])
+    3 -> EnumSet.of(values[0], values[1], values[2])
+    4 -> EnumSet.of(values[0], values[1], values[2], values[3])
+    5 -> EnumSet.of(values[0], values[1], values[2], values[3], values[4])
+    else -> EnumSet.copyOf(values.toSet())
+  }
+}
```

Comment:
There is no need to do this. Just use a regular set and mutable set in common code. We're no longer gaining any advantage of `EnumSet` having to go through this allocation-heavy path.

Reply:
[d7c57e4](https://github.com/square/kotlinpoet/pull/2006/commits/d7c57e4058efa53c4194db2384402f2653d0cb5f)